File: src/main/java/org/example/kalkulationsprogramm/domain/Arbeitsgang.java
package org.example.kalkulationsprogramm.domain;

import jakarta.persistence.*;
import lombok.Getter;
import lombok.Setter;

import java.util.ArrayList;
import java.util.List;

@Setter
@Getter
@Entity
public class Arbeitsgang {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id; // Original: ArbeitsgangID

    /**
     * Optimistisches Sperren: schuetzt gegen paralleles Speichern (auch aus
     * der Mobile-App, die keine Sperr-Oberflaeche hat).
     */
    @Version
    private Long version;

    @Column(nullable = false, unique = true)
    private String beschreibung;

    @ManyToOne(fetch = FetchType.LAZY, optional = false)
    @JoinColumn(name = "abteilung_id", nullable = false)
    private Abteilung abteilung;

    @OneToMany(mappedBy = "arbeitsgang", cascade = CascadeType.ALL, orphanRemoval = true)
    private List<Zeitbuchung> ZeitbuchungList = new ArrayList<>();

    @OneToMany(mappedBy = "arbeitsgang", cascade = CascadeType.ALL, orphanRemoval = true)
    private List<ArbeitsgangStundensatz> stundensaetze = new ArrayList<>();
}

File: src/main/kotlin/org/example/kalkulationsprogramm/domain/Arbeitsgang.kt
package org.example.kalkulationsprogramm.domain

import jakarta.persistence.*

@Entity
open class Arbeitsgang {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    open var id: Long? = null

    @Column(nullable = false, unique = true)
    open var beschreibung: String? = null

    @ManyToOne(fetch = FetchType.LAZY, optional = false)
    @JoinColumn(name = "abteilung_id", nullable = false)
    open var abteilung: Abteilung? = null

    @OneToMany(mappedBy = "arbeitsgang", cascade = [CascadeType.ALL], orphanRemoval = true)
    open var ZeitbuchungList: MutableList<Zeitbuchung> = ArrayList()

    @OneToMany(mappedBy = "arbeitsgang", cascade = [CascadeType.ALL], orphanRemoval = true)
    open var stundensaetze: MutableList<ArbeitsgangStundensatz> = ArrayList()

}